src/vs/workbench/contrib/preferences/browser/preferencesSearch.ts

```diff
@@ -305,4 +297,103 @@ export class SettingMatches {
 	}
 }
 
+class RemoteSearchProviderKeyCache {
```

Comment:
This doesn't really seem like a "cache", it just presents the same data from the ISettingsEditorModel in a different shape. Should this just be available from the settings editor model directly somehow?

Reply:
Considering it's only used for search, I'll leave the implementation here for now and will rename the class.